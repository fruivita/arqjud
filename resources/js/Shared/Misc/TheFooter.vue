<!--
    Rodapé da aplicação.

    @link https://vuejs.org/guide/introduction.html
    @link https://tailwindcss.com/docs
    @link https://inertiajs.com/
 -->
<script setup>
import Icone from '@/Shared/Icones/Icone.vue';
import { useDadosEstaticosStore } from '@/Stores/DadosEstaticosStore';
import { useTranslationsStore } from '@/Stores/TranslationsStore';

const __ = useTranslationsStore().__;
</script>

<template>
    <footer class="bg-primaria-100 px-3 py-6 text-center text-sm font-bold dark:bg-secundaria-800">
        <div class="space-y-3">
            <p>
                {{
                    `${useDadosEstaticosStore().appNome} - ${
                        useDadosEstaticosStore().appNomeCompleto
                    }`
                }}
            </p>

            <div class="flex items-center justify-center space-x-3">
                <a class="space-x-1" href="https://github.com/fruivita/arqjud/wiki">
                    <Icone class="inline h-6 w-6" nome="book" />

                    <span class="hover:underline">{{ __('Documentação') }}</span>
                </a>

                <a class="space-x-1" href="https://github.com/fruivita/arqjud">
                    <Icone class="inline h-6 w-6" nome="git" />

                    <span class="hover:underline">{{ useDadosEstaticosStore().appVersao }}</span>
                </a>
            </div>
        </div>
    </footer>
</template>
