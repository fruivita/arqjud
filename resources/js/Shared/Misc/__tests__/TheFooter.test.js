/**
 * Testes para o componente TheFooter.
 *
 * @link https://vuejs.org/guide/scaling-up/testing.html
 * @link https://test-utils.vuejs.org/guide/
 * @link https://vitest.dev/
 * @link https://stackoverflow.com/questions/69919632/how-to-test-vue3-and-intertia-with-jest
 */

import TheFooter from '@/Shared/Misc/TheFooter.vue';
import { createTestingPinia } from '@pinia/testing';
import { mount } from '@vue/test-utils';
import { beforeEach, describe, expect, test, vi } from 'vitest';

vi.stubGlobal('_dados', {
    app_nome: 'Foo',
    app_nome_completo: 'Foo Bar',
    app_versao: '1.0.0',
});

let mountFunction;

beforeEach(() => {
    const pinia = () => createTestingPinia({ stubActions: false });

    mountFunction = (options = {}) => {
        return mount(TheFooter, {
            global: {
                plugins: [pinia()],
            },
            ...options,
        });
    };
});

describe('TheFooter', () => {
    test('renderiza o componente respeitando o snapshot', () => {
        expect(mountFunction().html()).toMatchSnapshot();
    });
});
