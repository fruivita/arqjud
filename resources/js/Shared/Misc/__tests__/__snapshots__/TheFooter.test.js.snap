// Vitest Snapshot v1

exports[`TheFooter > renderiza o componente respeitando o snapshot 1`] = `
"<footer class=\\"bg-primaria-100 px-3 py-6 text-center text-sm font-bold dark:bg-secundaria-800\\">
  <div class=\\"space-y-3\\">
    <p>Foo - Foo Bar</p>
    <div class=\\"flex items-center justify-center space-x-3\\"><a class=\\"space-x-1\\" href=\\"#\\"><svg class=\\"inline h-6 w-6\\" fill=\\"currentColor\\" viewBox=\\"0 0 16 16\\" xmlns=\\"http://www.w3.org/2000/svg\\">
          <path d=\\"M1 2.828c.885-.37 2.154-.769 3.388-.893 1.33-.134 2.458.063 3.112.752v9.746c-.935-.53-2.12-.603-3.213-.493-1.18.12-2.37.461-3.287.811V2.828zm7.5-.141c.654-.689 1.782-.886 3.112-.752 1.234.124 2.503.523 3.388.893v9.923c-.918-.35-2.107-.692-3.287-.81-1.094-.111-2.278-.039-3.213.492V2.687zM8 1.783C7.015.936 5.587.81 4.287.94c-1.514.153-3.042.672-3.994 1.105A.5.5 0 0 0 0 2.5v11a.5.5 0 0 0 .707.455c.882-.4 2.303-.881 3.68-1.02 1.409-.142 2.59.087 3.223.877a.5.5 0 0 0 .78 0c.633-.79 1.814-1.019 3.222-.877 1.378.139 2.8.62 3.681 1.02A.5.5 0 0 0 16 13.5v-11a.5.5 0 0 0-.293-.455c-.952-.433-2.48-.952-3.994-1.105C10.413.809 8.985.936 8 1.783z\\"></path>
        </svg><span class=\\"hover:underline\\">Documentação</span></a><a class=\\"space-x-1\\" href=\\"#\\"><svg class=\\"inline h-6 w-6\\" fill=\\"currentColor\\" viewBox=\\"0 0 16 16\\" xmlns=\\"http://www.w3.org/2000/svg\\">
          <path d=\\"M15.698 7.287 8.712.302a1.03 1.03 0 0 0-1.457 0l-1.45 1.45 1.84 1.84a1.223 1.223 0 0 1 1.55 1.56l1.773 1.774a1.224 1.224 0 0 1 1.267 2.025 1.226 1.226 0 0 1-2.002-1.334L8.58 5.963v4.353a1.226 1.226 0 1 1-1.008-.036V5.887a1.226 1.226 0 0 1-.666-1.608L5.093 2.465l-4.79 4.79a1.03 1.03 0 0 0 0 1.457l6.986 6.986a1.03 1.03 0 0 0 1.457 0l6.953-6.953a1.031 1.031 0 0 0 0-1.457\\"></path>
        </svg><span class=\\"hover:underline\\">1.0.0</span></a></div>
  </div>
</footer>"
`;
